test/suite/test_prepare03.py

```diff
@@ -0,0 +1,211 @@
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+
+import wiredtiger, wttest
+from wtscenario import make_scenarios
+
+# test_prepre03.py
+#    Prepare transaction check post conditions for cursor operations
+
+# Pattern of test script is to invoke cursor operations in prepared
+# transactions to ensure they fail and to commit transaction and repeat
+# same operations to ensure normally they pass.
+class test_prepre03(wttest.WiredTigerTestCase):
+    """
+    Test basic operations
+    """
+    table_name1 = 'test_prepare_cursor'
+    nentries = 10
+
+    scenarios = make_scenarios([
+        ('file-col', dict(tablekind='col',uri='file')),
+        ('file-fix', dict(tablekind='fix',uri='file')),
+        ('file-row', dict(tablekind='row',uri='file')),
+        ('lsm-row', dict(tablekind='row',uri='lsm')),
+        ('table-col', dict(tablekind='col',uri='table')),
+        ('table-fix', dict(tablekind='fix',uri='table')),
+        ('table-row', dict(tablekind='row',uri='table'))
+    ])
+
+    def genkey(self, i):
+        if self.tablekind == 'row':
+            return 'key' + str(i)
+        else:
+            return long(i+1)
+
+    def genvalue(self, i):
+        if self.tablekind == 'fix':
+            return int(i & 0xff)
+        else:
+            return 'value' + str(i)
+
+    def assertCursorHasNoKeyValue(self, cursor):
+        keymsg = '/requires key be set/'
+        valuemsg = '/requires value be set/'
+        self.assertRaisesWithMessage(
+            wiredtiger.WiredTigerError, cursor.get_key, keymsg)
+        self.assertRaisesWithMessage(
+            wiredtiger.WiredTigerError, cursor.get_value, valuemsg)
+
+    def session_create(self, name, args):
+        """
+        session.create, but report errors more completely
+        """
+        try:
+            self.session.create(name, args)
+        except:
+            print('**** ERROR in session.create("' + name + '","' + args + '") ***** ')
+            raise
+
+    # Create and session and test cursor operations.
+    def test_prepare_cursor(self):
+        tablearg = self.uri + ':' + self.table_name1
+        preparemsg = "/ not permitted in a prepared transaction/"
+        if self.tablekind == 'row':
+            keyformat = 'key_format=S'
+        else:
+            keyformat = 'key_format=r'  # record format
+        if self.tablekind == 'fix':
+            valformat = 'value_format=8t'
+        else:
+            valformat = 'value_format=S'
+        create_args = keyformat + ',' + valformat
+
+        self.pr('creating session: ' + create_args)
+        self.session_create(tablearg, create_args)
+        self.pr('creating cursor')
+        cursor = self.session.open_cursor(tablearg, None, None)
+        self.assertCursorHasNoKeyValue(cursor)
+        self.assertEqual(cursor.uri, tablearg)
+
+        # Check insert and reset operations
+        for i in range(0, self.nentries):
```

Comment:
While 10 is pretty small is there a reason (I see it is used below for next/prev) to check the error case every iteration? If it works the first insert, it should work on all the later ones too.